common/nalreader_unittest.cpp

```diff
@@ -76,13 +76,13 @@ NALREADER_TEST(ReadAsWhole) {
     const uint8_t* nal;
     int32_t size;
 
-    NalReader reader(&g_data[0], g_data.size(), true);
+    NalReader reader(&g_data[0], g_data.size(), 0, true);
 
     EXPECT_TRUE(reader.read(nal, size));
-    EXPECT_EQ((size_t)size, g_data.size());
+    EXPECT_EQ((size_t)(size + 4), g_data.size());
 
     for (int32_t i(0); i < size; ++i)
-        EXPECT_EQ(nal[i], g_data[i]);
+        EXPECT_EQ(nal[i], g_data[i + 4]);
 
     EXPECT_FALSE(reader.read(nal, size));
 }
```

Comment:
Define `const int32_t offset = 4; // 1 initial junk byte + first start code`
Then replace 4's above with `offset`.

Add comment to test case about expectation, for example: "For whole read, the NalReader returns entire buffer contents after first start code.  That is, the first start code prefix is not included in the result."